### PR TITLE
Add runtime assertions for prim values

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107939

### CODE REVIEW

File: torch/_export/passes/add_runtime_assertions_for_constraints_pass.py [hunk 1]
```diff
@@ -226,6 +229,15 @@ def _insert_specialized_shape_assert_inplace(
         with graph.inserting_after(tensor_eq_node):
             _ = graph.call_function(torch.ops.aten._assert_async.msg, (tensor_eq_node, assert_msg))
 
+    def _insert_prim_assert_inplace(self, graph, node: torch.fx.Node, value: Any):
+        assert_msg = f"Input {node.name} is specialized at {value}"
```
Comment (gmagogsfm) [inline torch/_export/passes/add_runtime_assertions_for_constraints_pass.py:233]:
```suggestion
        assert_msg = f"Input {node.name} is specialized to be {value} at tracing time, it is not supported to pass in a different value at run time."
```

File: torch/_export/passes/add_runtime_assertions_for_constraints_pass.py [hunk 2]
```diff
@@ -226,6 +229,15 @@ def _insert_specialized_shape_assert_inplace(
...
+        with graph.inserting_after(node):
+            eq_node = graph.call_function(operator.eq, (node, value))
+        with graph.inserting_after(eq_node):
+            tensor_eq_node = graph.call_function(torch.ops.aten.scalar_tensor.default, (eq_node,))
```
Comment (gmagogsfm) [inline torch/_export/passes/add_runtime_assertions_for_constraints_pass.py:237]: Does this work for `str`?

Comment (tugsbayasgalan) [reply torch/_export/passes/add_runtime_assertions_for_constraints_pass.py:237]: Yeah I have a test case for it.

**Review by gmagogsfm — APPROVED** [review]
assuming you address nit comments

**Comment by avikchaudhuri** [comment]
Fix test failures, all of them are some assertion failure in test_upgrade. Maybe the test needs to be fixed?

**Pull request closed without merging**